atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClientImpl.java

```diff
@@ -139,23 +141,26 @@ public void close() {
             Executor executor,
             RowStreamAccumulator<V> rowStreamAccumulator) {
         return Futures.transformAsync(
-                session.executeAsync(executableStatement),
+                AtlasFutures.tracingFuture(() -> session.executeAsync(executableStatement), "CqlClientImpl: execute"),
```

Comment:
Btw, this is why I suggested you just create a specialized Executor, so you don't need to do this manually everywhere.

Reply:
So let's switch to that

Reply:
Canonically, we want something like:

```
DeferredTracer deferredTracer = new DeferredTracer();
        Executor executor = command -> executorService.execute(() -> {
            deferredTracer.withTrace(() -> { command.run(); return null;});
        });
        return Futures.whenAllSucceed(inputToListenableFutureMap.values())
                .call(() -> KeyedStream.stream(inputToListenableFutureMap)
                                .map(AtlasFutures::getDone)
                                .filter(Optional::isPresent)
                                .map(Optional::get)
                                .collectToMap(),
                        executor);
```

Reply:
Refactored to this structure.